build/mergeallmodules.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+#Assumes that ./x64 and ./x86 exist and contain the arch-specific copies of a DSC resource module
+
+set -e
+
+zip -L >/dev/null 2>&1
+if [ $? != 0 ]; then
+        echo "zip must be installed"
+        exit 1
+fi
+
+unzip -Z >/dev/null 2>&1
+if [ $? != 0 ]; then
+        echo "unzip must be installed"
+        exit 1
+fi
+
+TargetDir="./target"
+x64dir="./x64"
+x86dir="./x86"
+TestSigningDir=$1
+
+if [ "$TestSigningDir" = "" ]
+then
+	TestSigningDir="../test/config/testsigning"
+fi
+
+if [ -d "$TestSigningDir" ]
+then
+	SigningKeyFilePath="$TestSigningDir/signingkeys.gpg"
+	SigningKeyPassphrase="$TestSigningDir/passphrase"
+
+	if [ ! -f "$SigningKeyFilePath" ]
+	then
+		echo "GPG key file is not found"
+		exit 1
+	fi
+
+	if [ ! -f "$SigningKeyPassphrase" ]
+	then
+		echo "GPG key passphrase is not found"
+		exit 1
+	fi
+else
+	echo "Signing key directory doesn't exit. $TestSigningDir"
+        exit 1
+fi
+
+if [ -d "$TargetDir" ]
+then
+	rm -rf "$TargetDir"
+fi
+
+mkdir "$TargetDir"
+
+for ModuleFilePath in x64/*.zip;
+do
+        ModuleFileName=`echo $ModuleFilePath | xargs -n 1 basename`
+	ModuleName=`echo $ModuleFileName | cut -d _ -f 1`
+	Version=`echo $ModuleFileName | cut -d _ -f 2 |cut -d . -f 1,2`
```

Comment:
Gosh, this seems fragile, but if you need the ModuleName, you need it. I wonder, though, if you could get away without cutting the version information? If you need this, then please add a comment as to the exact format of the filename that you expect, thanks.

Note that `Version` doesn't appear to be used anywhere, so you can likely delete it.